dev/bots/analyze.dart

```diff
@@ -17,34 +18,52 @@ final String dart = path.join(flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', Pl
 final String pub = path.join(flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', Platform.isWindows ? 'pub.bat' : 'pub');
 final String pubCache = path.join(flutterRoot, '.pub-cache');
 
+class ExitException implements Exception {
+  ExitException(this.exitCode);
+
+  final int exitCode;
+
+  @alwaysThrows
```

Comment:
This annotation seems odd since io_internals.exit doesn't really throw?

I guess this is just here to say that code after calling this method never executes? Maybe add a comment?

Reply:
yeah i wasn't sure. it probably is fine to not include it at all... i was trying to be clever